tls/s2n_cipher_suites.c

```diff
@@ -1157,7 +1157,7 @@ int s2n_set_cipher_as_client(struct s2n_connection *conn, uint8_t wire[S2N_TLS_C
 
 static int s2n_wire_ciphers_contain(const uint8_t *match, const uint8_t *wire, uint32_t count, uint32_t cipher_suite_len)
 {
-    for (uint32_t i = 0; i < count; i++) {
+    for (size_t i = 0; i < count; i++) {
```

Comment:
This should probably be a `uint32_t`. I'm concerned about the `i < count` comparison: before the change the types matched, now they don't. 

Is this broken by `-Wsign-compare-check`? What line does the compiler warn about if so?